service/qos/service_level_controller.cc

```diff
@@ -641,7 +645,9 @@ future<scheduling_group> service_level_controller::auth_integration::get_user_sc
 }
 
 future<scheduling_group> service_level_controller::get_user_scheduling_group(const std::optional<auth::authenticated_user>& usr) {
-    SCYLLA_ASSERT(_auth_integration != nullptr);
```

Comment:
As I discussed with you earlier, another option is to keep the assertions, but only for cases where `_auth_integration` is really required. IIUC, these are the cases where the user is non-anonymous, and for maintenance socket the user is always anonymous. So, here and in `with_user_service_leve` it would be
```cpp
SCYLLA_ASSERT(_auth_integration != nullptr || !usr || !usr->name);
```
and the assertions in `find_effective_service_level` and `find_cached_effective_service_level` can actually stay the same because these functions are called only for non-anonymous users.

Moreover, these assertion changes should be temporary. It looks like the fix for #20813 will include starting `auth_service` before maintenance socket.